src/manifests/config-map.yaml

```diff
@@ -2,7 +2,7 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
-  name: {{ name }}
+  name: {{ name }}-charm
```

Comment:
Why are we adding this?

Reply:
For this configmap specifically, we dont need this.  

The `-charm` was added to the name of the deployment created by the charm (formerly `deployment/dex-auth`) because it was similarly named to the StatefulSet that juju creates (`statefulset/dex-auth`) which made it really easy to misunderstand who owned the broken pod (`pod/dex-auth-randomCharacters-0`).  To be consistent across the charm, I added the `-charm` to all objects created by charm code (if obj has `-charm` it came from charm code, if it doesn't it likely came from `juju/operator`).  

We can revert if needed, but I think we shouldn't.  Juju automatically creates some objects that have just `{{ name }}` as their name - us creating similarly named things is dangerous.  For example, it is easy to accidentally overwrite a ServiceAccount or reference a StatefulSet we don't mean to.  Including a suffix here seems prudent.

Reply:
Okay, that makes sense but we have to be careful with this when passing service names to other charms.

Reply:
yeah that's a good point.  That's a good argument for handling this directly in the charm rather than in the template